client/src/main/scala/io/delta/sharing/client/DeltaSharingProfileProvider.scala

```diff
@@ -17,22 +17,136 @@
 package io.delta.sharing.client
 
 import java.nio.charset.StandardCharsets.UTF_8
+import java.util.Locale
 
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.{DeserializationContext, JsonDeserializer}
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import com.fasterxml.jackson.databind.node.ObjectNode
 import org.apache.commons.io.IOUtils
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.Path
 import org.apache.spark.delta.sharing.TableRefreshResult
 
 import io.delta.sharing.client.util.JsonUtils
 
-case class DeltaSharingProfile(
-    shareCredentialsVersion: Option[Int],
-    endpoint: String = null,
-    bearerToken: String = null,
-    expirationTime: String = null)
+@JsonDeserialize(using = classOf[DeltaSharingProfileDeserializer])
+sealed trait DeltaSharingProfile {
+  val shareCredentialsVersion: Option[Int]
+  val endpoint: String
+}
+
+case class BearerTokenDeltaSharingProfile(override val shareCredentialsVersion: Option[Int],
+                                          override val endpoint: String = null,
```

Comment:
nit: fix the indent here and else where.

Reply:
I think the indent is correct and style check is passing already.